File: Android/TestSuperModel/app/src/main/java/com/soonzik/testsupermodel/Concert.java
package com.soonzik.testsupermodel;

import org.json.JSONObject;

import java.util.Date;

/**
 * Created by kevin_000 on 21/03/2015.
 */
public class Concert extends ActiveRecord {
    private int id = -1;
    private Address address = null;
    private Date date = null;

    public Concert() {}

    public Concert(JSONObject json) {
        super.createInstance(this, json, this.getClass());
    }

    @Override
    public String toString() {
        return (
                "id = " + Integer.toString(id)
                + " : Address = { " + (address != null ? address.toString() : "") + " }"
                + " : Date = " + (date != null ? date.toString() : "")
                );
    }
}

File: iOS SoonZik/SoonZik/Models/Concert.m
//
//  Concert.m
//  SoonZik
//
//  Created by Maxime Sauvage on 12/09/2014.
//  Copyright (c) 2014 Coordina. All rights reserved.
//

#import "Concert.h"

@implementation Concert

- (id)initWithJsonObject:(NSDictionary *)json
{
    self = [super init];
    
    NSLog(@"json : %@", json);
    
    self.identifier = [[json objectForKey:@"id"] intValue];
    NSDateFormatter *dateformat = [[NSDateFormatter alloc] init];
    [dateformat setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
    dateformat.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:1];
    self.date = [dateformat dateFromString:[json objectForKey:@"planification"]];
    self.url = [json objectForKey:@"url"];
    self.artist = [[User alloc] initWithJsonObject:[json objectForKey:@"user"]];
    self.address = [[Address alloc] initWithJsonObject:[json objectForKey:@"address"]];
    self.numberOfLikes = [[json objectForKey:@"likes"] integerValue];
    self.isLiked = [[json objectForKey:@"hasLiked"] boolValue];
    
    return self;
}

@end
